src/pymoca/tree.py

```diff
@@ -259,6 +290,1047 @@ def handle_walk(self, listener: TreeListener, tree: Union[ast.Node, dict, list])
+                sym_mod = ast.ClassModification()
+            for arg in instance.modification_environment.arguments:
+                if isinstance(arg.value, ast.ElementModification):
+                    if arg.value.component.indices != [[None]]:
+                        raise ModelicaSemanticError("Subscripting modifiers is not allowed.")
+                    if len(arg.value.component.child):
+                        # Move component reference down a level and apply modification to symbol
+                        # Don't stomp on original that may be used elsewhere
+                        arg = copy.copy(arg)
+                        arg.value = copy.copy(arg.value)
+                        arg.value.component = arg.value.component.child[0]
+                        sym_mod.arguments.append(arg)
+                    else:
+                        for sub_arg in arg.value.modifications:
+                            if isinstance(sub_arg, ast.ClassModification):
+                                sym_mod.arguments += sub_arg.arguments
+                            else:
+                                # Value modification - apply to symbol
+                                # Don't stomp on original that may be used elsewhere
+                                sub_arg = copy.copy(arg)
+                                sub_arg.value = copy.copy(arg.value)
+                                sub_arg.value.component = ast.ComponentRef(name="value")
+                                sym_mod.arguments.append(sub_arg)
+                else:
+                    raise UnimplementedError(f"{arg.value.__class__} symbol modification")
+
+            instance.modification_environment = sym_mod
+
+        # Set modification argument scope now that we have an instance
+        for index, arg in enumerate(instance.modification_environment.arguments):
+            if arg.scope is None:
+                # Make a copy so we don't change original AST or same arg used elsewhere
+                new_arg = copy.copy(arg)
+                new_arg.scope = instance.parent
+                instance.modification_environment.arguments[index] = new_arg
+
+        return instance
+
+    def _instantiate_parents_partially(
+        self,
+        class_: ast.InstanceClass,
+    ) -> None:
+        ALREADY_CALLED_VAR = "_instantiate_parents_partially_in_progress"
+
+        # Use a static variable to detect the resursion
+        if hasattr(self, ALREADY_CALLED_VAR):
+            # Continue the recursion upwards
+            self._instantiate_parents_partially_helper(class_)
+        else:
+            setattr(self, ALREADY_CALLED_VAR, True)
+
+            # Call below results in a different `class_.root` than `self`
+            self._instantiate_parents_partially_helper(class_)
+
+            # And now we merge the tree. Note that we then only do this once,
+            # for the first call to this function. Also update the parent refs.
+            self.extend(class_.root)
```

Comment:
Now that you have fixed this mess, I think we can move this back under `Tree` where it belongs.

Reply:
Wasn't it always part of `ast.Class`?